coalahtml/_coalahtml/app/controllers/files.js

```diff
@@ -33,7 +33,31 @@ angular.module('coalaHtmlApp')
     };
     $scope.fileName = $routeParams.fileName || $rootScope.ROOTHOME;
     parseCoalaProject($scope.fileName);
-    $scope.basename = function(path_name){
+
+    $scope.basename = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
       return path_name.split('/').slice(-1)[0];
     };
+    var len = function(path_name) {
+      path_name = path_name || $rootScope.ROOTHOME;
+      return path_name.split('/').length;
+    };
+    $scope.getNameLink = function(fileName) {
+      var linkNameContainer = [], files = fileName.split('/');
+      var base = files.slice(0, len($rootScope.ROOTHOME)).join('/');
```

Comment:
Isn't `files.slice(0, len($rootScope.ROOTHOME)).join('/')` the same as `$rootScope.ROOTHOME`?